numpy/lib/npyio.py

```diff
@@ -2242,9 +2243,14 @@ def encode_unicode_cols(row_tup):
     if usemask:
         output = output.view(MaskedArray)
         output._mask = outputmask
+    output = np.squeeze(output)
     if unpack:
-        return output.squeeze().T
-    return output.squeeze()
+        if names is not None and len(names) > 1:
```

Comment:
Why the `len(names) > 1` condition?

Reply:
If the goal is to match `squeeze`, then `len(names) != 1` would work better for when `names == []`.

Reply:
I can change it, but I think `names == []` necessarily implies `output` is empty (i.e. the input was an empty file).

Consider the following two calls:
```
np.genfromtxt(StringIO(''), dtype=None, unpack=True)
np.genfromtxt(StringIO(''), dtype={'names': (), 'formats': ()}, unpack=True)
```
With `len(names > 1)`, both will return an empty `ndarray` , but with `len(names) != 1)`, both will return an empty list.